internal/queue/queue_test.go

```diff
@@ -172,10 +173,7 @@ func TestDequeueBlock(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	select {
-	case err := <-errChan:
-		if err != nil {
-			t.Fatal(err)
-		}
+	if err := <-errChan; err != nil {
```

Comment:
Oh duh. This used to wait on a done chan and the error but never removed the select. Good linter 🙂. Also really annoyed these aren't caught on PR for whatever reason. We should really dedicate time to find why it lets some slip

Reply:
yeah, weird this wasn't caught by original PR...

Reply:
Oh geeze, no I remember what we should've done in this test. If there's a bug in dequeue and we never return we'll sit in the receive for the errChan potentially forever

Reply:
Could you add back in the select and additionally wait on a time.After() of 10 seconds or so?

Reply:
Perfect, thanks!